src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlStoreClient.cs

```diff
@@ -129,21 +129,18 @@ public async Task<IReadOnlyList<ResourceWrapper>> GetAsync(IReadOnlyList<Resourc
             return resources;
         }
 
-        private static Lazy<string> ReadRawResource(SqlDataReader reader, Func<MemoryStream, string> decompress, int index)
+        private static string ReadRawResource(SqlDataReader reader, Func<MemoryStream, string> decompress, int index)
         {
             var rawResourceBytes = reader.GetSqlBytes(index).Value;
-            Lazy<string> rawResource;
+            string rawResource;
             if (rawResourceBytes.Length == 1 && rawResourceBytes[0] == 0xF) // invisible resource
             {
-                rawResource = new Lazy<string>(_invisibleResource);
+                rawResource = _invisibleResource;
             }
             else
             {
-                rawResource = new Lazy<string>(() =>
-                {
-                    using var rawResourceStream = new MemoryStream(rawResourceBytes);
-                    return decompress(rawResourceStream);
-                });
+                using var rawResourceStream = new MemoryStream(rawResourceBytes);
```

Comment:
Might be interesting to look at RecyclableMemoryStream here

Reply:
I don't think there is a problem with it, as entire data reader execution is very fast. We are spending time elsewhere...